gateway/src/main/java/io/camunda/zeebe/gateway/impl/stream/ClientStreamAdapter.java

```diff
@@ -46,6 +64,17 @@ public void handle(
     responseObserver.setOnCancelHandler(removeJobStream);
   }
 
+  private void handleError(
+      final ServerCallStreamObserver<ActivatedJob> responseObserver,
+      final String field,
+      final String expectation,
+      final String actual) {
```

Comment:
:wrench: You can make `actual` an `Object`. For objects (e.g. `String`) this will change nothing, but for primitives, this means we're now boxing them (e.g. `long` becomes `Long`). This can sometimes be a performance thing, but since we do this only in the error case here, I think simplicity is easier.

If you do, then you don't need to convert the parameters above (e.g. `String.format("'%d'", request.getTimeout())`. You can just pass `request.getTimeout()`, and `request.getType()`.

When it formats `%s` with an object, it'll just call its `toString` method :+1: 

By the way, since Java 16 (? maybe earlier, can't remember), you can do:

```java
    final var format = "Expected to stream activated jobs with %s to be %s, but it was %s";
    final String errorMessage = format.formatted(field, expectation, actual);
    responseObserver.onError(
        new StatusRuntimeException(Status.INVALID_ARGUMENT.withDescription(errorMessage)));
```

Reply:
Wouldn't it reduce the readability of the code if we pass `Object`? I think, it is too much of optimisation that we will start to sacrifice readability. WDYT?